schematic/models/validate_attribute.py

```diff
@@ -2027,13 +2028,18 @@ def filename_validation(
             errors: list[str] Error details for further storage.
             warnings: list[str] Warning details for further storage.
         """
+
+        if dataset_scope is None:
```

Comment:
If this cannot be none I would expect that the argument of the function is not optional (ie: No `= None`).

Could you also add this to the Args of the docstring?

Reply:
@BryanFauble yeah that makes sense thanks for catching